File: java/19. Remove Nth Node From End of List/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {

        ListNode preHead = new ListNode(-1, head);

        ListNode current = head;
        int count = 0;

        while (current != null) {
            current = current.next;
            count++;
        }

        int turn = count - n;
        ListNode prev = preHead;
        current = head;

        for (int i = 0; i < turn; i++) {
            prev = prev.next;
            current = current.next;
        }

        prev.next = current.next;

        return preHead.next;

    }
}

File: python/19. Remove Nth Node From End of List/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        preHead = ListNode(next = head)

        fast = preHead
        slow = preHead
        for i in range(n + 1):
            fast = fast.next

        while fast :
            fast = fast.next
            slow = slow.next

        slow.next = slow.next.next

        return preHead.next
     